dashboard/src/api/commonRequest.ts

```diff
@@ -0,0 +1,22 @@
+import type { AxiosRequestConfig } from 'axios';
+
+import type { ResponseData } from '@/types/general';
+
+import http from './api';
+
+const httpMap = {
+  get: http.get,
+  post: http.post,
+};
+
+export const makeRequest = async <T>(
+  url: string,
+  config?: AxiosRequestConfig,
+  variant: keyof typeof httpMap = 'get',
```

Comment:
```suggestion
  method: keyof typeof httpMap = 'get',
```
method is more suggestive since this refers to the http method used by the request